libs/platform/core/src/services/feature-options/default-feature-options.service.ts

```diff
@@ -5,14 +5,16 @@ import {
 } from './feature-options.service';
 
 export class DefaultFeatureOptionsService implements FeatureOptionsService {
-  protected options?: FeatureOptions;
+  protected options?: Record<string, object | undefined>;
 
   constructor(protected optionsArr = inject(FeatureOptions, null)) {
     this.options =
       optionsArr?.reduce((acc, options) => ({ ...acc, ...options }), {}) ?? {};
   }
 
-  getFeatureOptions(name: string): FeatureOptions {
+  getFeatureOptions<N extends keyof FeatureOptions>(name: N): FeatureOptions[N];
+  getFeatureOptions(name: string): object;
+  getFeatureOptions(name: string): object {
```

Comment:
```suggestion
  getFeatureOptions(name: string): object {
```

Reply:
We can't remove that overload version as it's not going to be compatible with the generic overload and implementation.
<img width="699" alt="image" src="https://user-images.githubusercontent.com/3644678/233093607-ed55171a-897a-4754-ae55-a37d6b84271c.png">